### PR TITLE
Add development deps to dependencies.yaml

### PR DESCRIPTION
**Pull request by cwharris**
## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

### CODE REVIEW

File: dependencies.yaml [hunk 1]
```diff
@@ -56,6 +57,14 @@ dependencies:
           - scikit-build=0.17.1
           - tritonclient=2.26 # Required by NvTabular, force the version, so we get protobufs compatible with 4.21
           - ucx=1.14
+  development_cpp:
+    common:
+      - output_types: [conda]
+        packages:
+          - clangdev=16
+          - include-what-you-use=0.18
```
Comment (mdemoret-nv) [inline dependencies.yaml:65]: IMPORTANT: I believe that IWYU 0.20 is required for clang 16

**Review by mdemoret-nv — APPROVED** [review]

File: dependencies.yaml (hunk 1 above)
Comment (dagardner-nv) [inline dependencies.yaml:65]: IMPORTANT: Be sure to update `docker/conda/environments/cuda11.8_dev.yml` to match.

Also locally make change a cpp file and then run `ci/scripts/cpp_checks.sh` to verify that its working. I believe I tried this a while ago and ran into some trouble.

**Pull request merged**